plugins/modules/ndo_l3out_secondary_ip.py

```diff
@@ -0,0 +1,578 @@
+  Use M(cisco.mso.ndo_template) to create the L3Out template.
+- The O(l3out) must exist before using this module in your playbook.
+  Use M(cisco.mso.ndo_l3out_template) to create the L3Out.
+- The parent object defined by O(parent_type) must exist before using this module in your playbook.
+  Use M(cisco.mso.ndo_l3out_routed_interface) to create the L3Out Routed Interface.
+  Use M(cisco.mso.ndo_l3out_routed_sub_interface) to create the L3Out Routed Sub-Interface.
+  Use M(cisco.mso.ndo_l3out_svi_interface) to create the L3Out SVI Interface.
+  Use M(cisco.mso.ndo_l3out_floating_svi_interface) to create the L3Out Floating SVI Interface.
+  Use M(cisco.mso.ndo_l3out_floating_svi_interface_path_attributes) to create the L3Out Floating SVI Interface Path Attributes.
+
+seealso:
+- module: cisco.mso.ndo_template
+- module: cisco.mso.ndo_l3out_template
+- module: cisco.mso.ndo_l3out_routed_interface
+- module: cisco.mso.ndo_l3out_routed_sub_interface
+- module: cisco.mso.ndo_l3out_svi_interface
+- module: cisco.mso.ndo_l3out_floating_svi_interface
+- module: cisco.mso.ndo_l3out_floating_svi_interface_path_attributes
+extends_documentation_fragment: cisco.mso.modules
+"""
+
+EXAMPLES = r"""
+- name: Create a secondary address on a L3out Routed Interface of type port
+  cisco.mso.ndo_l3out_secondary_ip:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: l3out_template
+    l3out: l3out_1
+    node_id: 101
+    path: eth1/3
+    parent_type: routed
+    secondary_address: 3::102/16
+    state: present
+
+- name: Create a secondary address on a L3out Routed Interface of type port channel
+  cisco.mso.ndo_l3out_secondary_ip:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: l3out_template
+    l3out: l3out_1
+    port_channel:
+      uuid: '{{ port_channel_interface_3.current.uuid }}'
+    parent_type: routed
+    secondary_address: 10.2.0.2/24
+    state: present
+
+- name: Create a secondary address on a L3out SVI Interface of type port
+  cisco.mso.ndo_l3out_secondary_ip:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: l3out_template
+    l3out: l3out_1
+    node_id: 101
+    path: eth1/1
+    encapsulation_type: vlan
+    encapsulation_value: 103
+    parent_type: svi
+    secondary_address: 10.0.0.2/24
+    state: present
+
+- name: Create a secondary address on a L3out SVI Interface of type port channel
+  cisco.mso.ndo_l3out_secondary_ip:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: l3out_template
+    l3out: l3out_1
+    port_channel:
+      uuid: '{{ port_channel_interface_1.current.uuid }}'
+    encapsulation_type: vxlan
+    encapsulation_value: 50001
+    parent_type: svi
+    secondary_address: 1::102/16
+    state: present
+
+- name: Create a secondary address on a L3out SVI Interface of type virtual port channel
```

Comment:
Shall we mention side a here if valid?
- name: Create a secondary address on a L3out SVI Interface of type virtual port channel side a